src/mappings/nabla/routerEventHandler.ts

```diff
@@ -0,0 +1,150 @@
+import {
+    decodeEvent,
+    Event_Swap,
+    Event_SwapPoolRegistered,
+    Event_SwapPoolUnregistered,
+} from '../../abi/router'
+import * as swapPoolAbi from '../../abi/swap'
+import {
+    getSwapPoolsOfRouterForToken,
+    getOrCreateNablaToken,
+    getSwapPool,
+} from './creation'
+import { updateSwapPoolCoverageAndSupply } from './swapPoolEventHandler'
+import { hexToSs58, ss58ToHex } from './addresses'
+
+export async function handleRouterEvent(
+    ctx: EventHandlerContext,
+    router: Router
+) {
+    const event = decodeEvent(ctx.event.args.data)
+
+    switch (event.__kind) {
+        case 'OwnershipTransferred':
+            // This event will always be emitted on router creation
+            // no action required
+            break
+
+        case 'Paused':
+            await handlePaused(ctx, router)
+            break
+
+        case 'Swap':
+            await handleSwap(ctx, event, router)
+            break
+
+        case 'SwapPoolRegistered':
+            await handleSwapPoolRegistered(ctx, event, router)
+            break
+
+        case 'SwapPoolUnregistered':
+            await handleSwapPoolUnregistered(ctx, event, router)
+            break
+
+        case 'Unpaused':
+            await handleUnpaused(ctx, router)
+            break
+    }
+}
+
+export async function handlePaused(ctx: EventHandlerContext, router: Router) {
+    // Set the 'paused' property to true
+    router.paused = true
+    await ctx.store.save(router)
+}
+
+export async function handleUnpaused(ctx: EventHandlerContext, router: Router) {
+    // Set the 'paused' property to false
+    router.paused = false
+    await ctx.store.save(router)
+}
+
+export async function handleSwap(
+    ctx: EventHandlerContext,
+    event: Event_Swap,
+    router: Router
+) {
+    const tokenIn = await getOrCreateNablaToken(ctx, event.tokenIn)
+    const tokenOut = await getOrCreateNablaToken(ctx, event.tokenOut)
+
+    const swapPoolIn = await getSwapPoolsOfRouterForToken(
+        ctx,
+        router.id,
+        tokenIn.id
+    )
+    const swapPoolOut = await getSwapPoolsOfRouterForToken(
+        ctx,
+        router.id,
+        tokenOut.id
+    )
+
+    if (swapPoolIn !== undefined) {
+        await updateSwapPoolCoverageAndSupply(ctx, swapPoolIn)
+        await ctx.store.save(swapPoolIn)
+    }
+
+    if (swapPoolOut !== undefined) {
+        await updateSwapPoolCoverageAndSupply(ctx, swapPoolOut)
+        await ctx.store.save(swapPoolOut)
+    }
+}
+
+export async function handleSwapPoolRegistered(
+    ctx: EventHandlerContext,
+    event: Event_SwapPoolRegistered,
+    router: Router
+) {
+    const swapPoolSs58Address = hexToSs58(event.pool)
+    const swapPool = await getSwapPool(ctx, swapPoolSs58Address)
+
+    if (swapPool === undefined) {
+        // this is a non-standard or malicious swap pool, ignore
+        return
+    }
+
+    const token = await getOrCreateNablaToken(ctx, ss58ToHex(swapPool.token.id))
```

Comment:
Shouldn't this variable be passed as a parameter when calling `getSwapPoolsOfRouterForToken` below? Otherwise, it looks like it's not used at all.

Reply:
You are right. Actually the result is not required and it just matters that the function is called. I changed it and [opened a new ticket](https://github.com/pendulum-chain/pendulum-squids/issues/63) to avoid these errors in the future.